src/lib/modules/query/hinter.js

```diff
@@ -17,31 +17,42 @@ class Hinter {
    * @param {Object} options
    * @param {Array<String>} options.collectionNames
    */
-  getHintsByValue(value, options) {
-    if (!value) return [];
+  getHintsByValue(fullQuery, currentWord, options) {
+    if (!fullQuery) return [];
     options = options && _.isObject(options) ? options : [];
 
     let hints = null;
+    let value = null;
 
-    value = value.trim ? value.trim() : value;
-    value = value.toLowerCase ? value.toLowerCase() : value;
+    value = fullQuery.trim ? fullQuery.trim() : fullQuery;
+    value = fullQuery.toLowerCase ? fullQuery.toLowerCase() : fullQuery;
 
-    let collectionName = parser.parseCollectionName(value);
-    // let functionName = parser.parseFunctionName(value);
+    let collectionName = parser.parseCollectionName(fullQuery);
+    let functionName = parser.parseFunctionName(fullQuery);
 
     var collectionNameRegex = collectionName ? new RegExp('[\\s\\S]+' + collectionName + '\.[\\s\\S]*') : null;
```

Comment:
I pulled down your PR and I'm finding that when I type a query like

```
db.cars.
```

the autocomplete continues to return the collection names for hints. I found some issues with these 2 regexes that seem to be causing the issue. Here are 2 newer versions:

```
var collectionNameRegex = collectionName ? new RegExp('[\\s\\S]+\.' + collectionName + '\\.[\\s\\S]*', 'i') : null;
    var collectionNamePlusFunctionRegex = collectionName ? new RegExp('[\\s\\S]+\.' + collectionName + '\\.[a-zA-Z0-9]*\\(', 'i') : null;
```

Do you mind putting these in and trying it out on your end?